planning.py

```diff
@@ -1346,6 +1344,160 @@ def result(problem, action):
+                else: 
+                    prefix = []
+                    outcome = copy.deepcopy(problem)
+                if j < len(plan.action)-1:
+                    suffix = plan.action[j+1:]
+                else: 
+                    suffix = []
+                for sequence in Problem.refinements(hla, outcome, hierarchy): # find refinements
+                    frontier.append(Angelic_Node(outcome.init, plan, prefix + sequence+ suffix, prefix+sequence+suffix))
+
+
+            n+=1
+
+    def intersects_goal(problem, reachable_set):
+        """
+        Find the intersection of the reachable states and the goal
+        """
+        return [y for x in list(reachable_set.keys()) for y in reachable_set[x] if all(goal in y for goal in problem.goals)] 
+
+
+    def is_primitive(plan,  library):
+        """
+        checks if the hla is primitive action 
+        """
+        for hla in plan.action: 
+            indices = [i for i, x in enumerate(library['HLA']) if expr(x).op == hla.name]
+            for i in indices:
+                if library["steps"][i]: 
+                    return False
+        return True
+             
+
+
+    def reach_opt(init, plan): 
+        """
+        Finds the optimistic reachable set of the sequence of actions in plan 
+        """
+        reachable_set = {0: [init]}
+        optimistic_description = plan.action #list of angelic actions with optimistic description
+        return Problem.find_reachable_set(reachable_set, optimistic_description)
+ 
+
+    def reach_pes(init, plan): 
+        """ 
+        Finds the pessimistic reachable set of the sequence of actions in plan
+        """
+        reachable_set = {0: [init]}
+        pessimistic_description = plan.action_pes # list of angelic actions with pessimistic description
+        return Problem.find_reachable_set(reachable_set, pessimistic_description)
+
+    def find_reachable_set(reachable_set, action_description):
+        """
+	Finds the reachable states of the action_description when applied in each state of reachable set.
+	"""
+        for i in range(len(action_description)):
+            reachable_set[i+1]=[]
+            if type(action_description[i]) is Angelic_HLA:
+                possible_actions = action_description[i].angelic_action()
+            else: 
+                possible_actions = action_description
+            for action in possible_actions:
+                for state in reachable_set[i]:
+                    if action.check_precond(state , action.args) :
+                        if action.effect[0] :
+                            new_state = action(state, action.args).clauses
+                            reachable_set[i+1].append(new_state)
+                        else: 
+                            reachable_set[i+1].append(state)
+        return reachable_set
+
+
+	
+    def making_progress(plan, initialPlan):
+        """ 
+        Not correct 
+        """
```

Comment:
It's good that you noted that this isn't working. Perhaps you can comment on what you think the problem is, or what would be needed to make it correct.

Reply:
I have commented on what should be the purpose of making_progress and what it is computing at the moment.